src/main/io/gps/gps_config.c

```diff
@@ -0,0 +1,47 @@
+
+#include "platform.h"
+
+#ifdef USE_GPS
+
+#include "gps_config.h"
+
+ubloxVersion_e ubloxDetectVersion(const char * szBuf, const uint8_t nBufSize) {
+    ubloxVersion_e ret = UNDEF;
+    // ublox_5   hwVersion 00040005
+    if (strncmp(szBuf, "00040005", nBufSize) == 0) {
+        ret = M5;
+    } else if (strncmp(szBuf, "00040007", nBufSize) == 0) {
+        ret = M6;
+    } else if (strncmp(szBuf, "00070000", nBufSize) == 0) {
+        ret = M7;
+    } else if (strncmp(szBuf, "00080000", nBufSize) == 0) {
+        ret = M8;
+    } else if (strncmp(szBuf, "00190000", nBufSize) == 0) {
+        ret = M9;
+    } else if (strncmp(szBuf, "000A0000", nBufSize) == 0) {
+        ret = M10;
+    }
+    return ret;
+}
+
+char* ubloxVersionToString(ubloxVersion_e version) {
+    switch (version) {
```

Comment:
There are better ways to translate a enum to text. Look at the second answer here: 
https://stackoverflow.com/questions/9907160/how-to-convert-enum-names-to-string-in-c/58500930#58500930